rust/src/file.rs

```diff
@@ -212,18 +218,24 @@ impl File {
     }
 
     fn truncate(&mut self) -> Result<()> {
-        match self.header.block_type {
+        let truncated_size = match self.header.block_type {
             BlockType::Raw => {
                 let header_size = self.header.size_on_disk();
                 let body_size = self.header.file_size();
                 let size = header_size + body_size;
-
-                match self.file.set_len(size as u64) {
-                    Ok(_) => Ok(()),
-                    Err(_) => Err("Could not truncate file")
-                }
+                size as u64
             },
-            _ => Ok(())
+            BlockType::LZ4 | BlockType::LZ4HC => {
+                let last_block_idx = self.header.file_vol() - 1;
+                let jump_table = &mut *self.header.jump_table.as_mut().unwrap();
```

Comment:
> In fact, this whole function doesn't modify the wkw::File at all. If you want, you could even improve the code by changing the function signature to fn truncate(&self).

 Ah, interesting, I assumed that a `mut` was necessary since `set_len` mutates the file, but I'm not fully acquainted with the borrow model yet, so that was a helpful hint, thanks!